### ISSUE TITLE
Download videos from youtube

### ISSUE THREAD

**Issue by BBloggsbott** [RipMeApp/ripme#675]
Found an [API](https://gitlab.com/axet/vget) to download videos from youtube. Can get it done easily.

**Comment by tehloxely** [comment]
How is it compared to the powerhouse [youtube-dl](https://github.com/rg3/youtube-dl)?

**Comment by BBloggsbott** [comment]
youtube-dl requires it to be installed in the system, and I think it is available only for linux system.

**Comment by Hrxn** [comment]
youtube-dl is cross-platform and does not need to be installed in any way.

**Comment by BBloggsbott** [comment]
Oh, I didn't know that. I guess youtube-dl can be used

**Comment by rautamiekka** [comment]
Yeah, it's true `youtube-dl`'s doc makes it sound like only for Linux, but there's not much difference when run on Window$: use double quotes around URLs for safety (Linux shells are less strict about them, but you should use single quotes anyway), and if you use a BAT script, which you should, you need to double up percentage signs.

Licensing wouldn't be a problem if RipMe would port code from `youtube-dl` since both are MIT, and YouTube doesn't change all that often, but you might not wanna sorta re-invent the wheel even that way.

**Comment by kevin51jiang** [comment]
We have talked in the past about making a wrapper for YouTube dl since that will accommodate for any changes YouTube makes, that's still a possibility if you want to do it. 

However, since YouTube dl still exists and works wonderfully on its own, we have put that rather low on the priority scale.

**Comment by BBloggsbott** [comment]
Yeah, but having everything in one application will be better right?

**Comment by cyian-1756** [comment]
> Yeah, but having everything in one application will be better right?

Not really. We'd end up redoing a bunch of work that the guys over at youtube-dl and end up with something that supports way fewer sites (Because we don't have the man power to support the same number of sites as youtube-dl)

We could try embedding youtube-dl and running it with jython, but jython doesn't look like it'll be getting support for python3 anytime soon and python2 is end of life

**Comment by shinji257** [comment]
Old discussion but I was checking to see if this was considered and Youtube-DL comes in binary form that could be used.  It wouldn't require Python to be installed for it to work.
